agent/engine/docker_task_engine.go

```diff
@@ -1126,7 +1131,18 @@ func (engine *DockerTaskEngine) State() dockerstate.TaskEngineState {
 
 // Version returns the underlying docker version.
 func (engine *DockerTaskEngine) Version() (string, error) {
-	return engine.client.Version()
+	if engine.dockerVersion != "" {
```

Comment:
Do you think we need a lock for this method? Also, do we need a retry for this?

Reply:
I don't think this needs a retry. Looking at the usage, there's a remote possibility of a race condition. I'll add a scoped down mutex just for this for now. We can expand it's scope to the whole of `DockerTaskEngine` if needed.